testworld/park.go

```diff
@@ -293,6 +297,14 @@ func (h *host) init() error {
 	if err != nil {
 		return err
 	}
+
+	if h.name == "Mallory" {
+		malloryDocMockSrv := new(mockdoc.MockService)
```

Comment:
Are you sure this is the actual doc service that gets called for the host? 
Seems though the underlying idea is correct, this would not inject the mock doc service correctly because the p2p bootstrapper you create below is not added to the host anywhere. 
My idea is that you can take the `h.bootstrappedCtx[bootstrap.BootstrappedPeer]` struct(type `p2p.peer`) here and set its `handlerCreator` field to something you control during the test.  It gets set for the host originally at https://github.com/centrifuge/go-centrifuge/blob/develop/p2p/bootstrapper.go#L44

Reply:
Hi @vimukthi-git 
you can try comment line 54 ,57 59 of signature_test.go, following is my result
![image](https://user-images.githubusercontent.com/6283645/58803046-27308c00-8641-11e9-9857-b5326e38b5f7.png)

Reply:
init() function is called before live()